enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -919,7 +901,15 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 
 		mu.Lock()
 		lastOp = op
-		taskID = op.GetName()
+
+		// If this is the first op received from Stream, update the
+		// execution status in MySQL.
+		if op.GetName() != taskID {
+			taskID = op.GetName()
+			if err := s.updateExecution(ctx, taskID, stage, lastOp); err != nil {
+				return err
+			}
+		}
```

Comment:
previously we'd skip this write if the execution took less than 5s. Does it seem worth trying to preserve that behavior?

Reply:
resolved offline